src/main/flight/pid.c

```diff
@@ -538,6 +567,79 @@ static void handleItermRotation()
     }
 }
 
+#ifdef USE_ACRO_TRAINER
+
+int8_t acroTrainerSign(float x)
+{
+    return (x > 0.0f) - (x < 0.0f);
+}
+
+// Acro Trainer - Manipulate the setPoint to limit axis angle while in acro mode
+// There are three states:
+// 1. Current angle has exceeded limit
+//    Apply correction to return to limit (similar to pidLevel)
+// 2. Future overflow has been projected based on current angle and gyro rate
+//    Manage the setPoint to control the gyro rate as the actual angle  approaches the limit (try to prevent overshoot)
+// 3. If no potential overflow is detected, then return the original setPoint
+
+static float applyAcroTrainer(int axis, const rollAndPitchTrims_t *angleTrim, float setPoint)
+{
+    float ret = setPoint;
+
+#ifndef SIMULATOR_BUILD
```

Comment:
Is this actually breaking for SITL, and if so, why?

Reply:
Travis was generating errors while compiling SITL and I may have overreacted with a sledgehammer approach.  Part of that was cause by the frustration from being unable to compile SITL on OS X.

I just took my local repo over to a Linux platform and it seems to compile SITL without any errors or warnings even without the `#ifndef SIMULATOR_BUILD` stuff so I don't understand.  The errors from Travis are:
```
./src/main/flight/pid.c: In function ‘pidController’:
./src/main/flight/pid.c:630:74: error: ‘projectedAngle’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
             DEBUG_SET(DEBUG_ACRO_TRAINER, 3, lrintf(projectedAngle * 10.0f));
                                                                          ^
./src/main/flight/pid.c:591:15: note: ‘projectedAngle’ was declared here
         float projectedAngle;
               ^
cc1: all warnings being treated as errors
```

I don't understand what's going on as the variable `projectedAngle` is not in `pidController()` - it's in the `applyAcroTrainer()` function called from `pidController()`.  So maybe something weird is happening with inlines.

Looking at `applyAcroTrainer()` the variable is declared without initialization because it's appropriate that the default value be 0.  I guess I could explicitly set it to zero but it seems like working around an invalid warning.

Reply:
Intersting. SITL is building fine here on ubuntu with `gcc version 7.3.0 (Ubuntu 7.3.0-16ubuntu3)`.

Reply:
Inlines are happening as intended here :)

Reply:
Yeah, not sure what's happening with Travis and SITL - compiles fine for me on Fedora 27 (gcc version 7.2.1 20170915 (Red Hat 7.2.1-2)) as well.  But Travis warns that variable "may" be unitialized.  Probably because a potential first use is `DEBUG_SET` and it is a macro.  Doing an explicit initialization to zero will probably make it happy.

`float projectedAngle = 0;`

Reply:
Well I meant to say that it will implicitly be 0 when used for the first time. I guess the chance that it will first be "used" in a macro is what's generating the compiler warning.  Generally the guidance has been to not bother initializing with the trivial implicit value but it seems in this case it's required.  I generally like to explicitly initialize my variables (carry-over from other languages) but have been consciously avoiding it for trivial cases as guidance and reviews on other PR's have suggested it isn't needed and just clutters.

Reply:
@etracer65: static and global variables are implicitly initialized to (binary) zeroes. But automatic variables are uninitialized, so this function can log random value that was on stack. Exactly what is the warning for ...

Reply:
There's no "trivial implicit value" whatsoever in this case, your variable remains uninitialized until first assignment and it's value will be indeterminate

Reply:
Ah that's the point - I was not thinking about the `static` distinction.  Makes sense now.  Still doesn't explain why it compiles locally with no warnings though. But I guess that's another matter.

Reply:
We've no `-Werror` locally I think